adapter/service/outlvlswitch/routing.go

```diff
@@ -0,0 +1,181 @@
+package outlvlswitch
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/futurehomeno/fimpgo"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/futurehomeno/cliffhanger/adapter"
+	"github.com/futurehomeno/cliffhanger/router"
+)
+
+// Constants defining routing service, commands and events.
+const (
+	CmdLvlSet       = "cmd.lvl.set"
+	CmdLvlGetReport = "cmd.lvl.get_report"
+	EvtLvlReport    = "evt.lvl.report"
+	CmdLvlStart     = "cmd.lvl.start"
+	CmdLvlStop      = "cmd.lvl.stop"
+	CmdBinarySet    = "cmd.binary.set"
+
+	OutLvlSwitch = "out_lvl_switch"
+)
+
+// RouteService returns routing for service specific commands.
+func RouteService(adapter adapter.Adapter) []*router.Routing {
+	return []*router.Routing{
+		RouteCmdLvlSet(adapter),
+		RouteCmdBinarySet(adapter),
+		RouteCmdLvlGetReport(adapter),
+	}
+}
+
+// RouteCmdLvlSet returns a routing responsible for handling the command.
+func RouteCmdLvlSet(adapter adapter.Adapter) *router.Routing {
+	return router.NewRouting(
+		HandleCmdLvlSet(adapter),
+		router.ForService(OutLvlSwitch),
+		router.ForType(CmdLvlSet),
+	)
+}
+
+// HandleCmdLvlSet returns a handler responsible for handling the command.
+func HandleCmdLvlSet(adapter adapter.Adapter) router.MessageHandler {
+	return router.NewMessageHandler(
+		router.MessageProcessorFn(func(message *fimpgo.Message) (*fimpgo.FimpMessage, error) {
+			s := adapter.ServiceByTopic(message.Topic)
+			if s == nil {
+				return nil, fmt.Errorf("adapter: service not found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			outLvlSwitch, ok := s.(Service)
+			if !ok {
+				return nil, fmt.Errorf("adapter: incorrect service found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			lvl, err := message.Payload.GetIntValue()
+			if err != nil {
+				return nil, fmt.Errorf("adapter: error while getting level value from message: %w", err)
+			}
+
+			err = getDurationAndSetLevel(message, outLvlSwitch, lvl)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: error while setting level: %w", err)
+			}
+
+			_, err = outLvlSwitch.SendLevelReport(true)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: error while sending level report: %w", err)
+			}
+
+			return nil, nil
+		}),
+	)
+}
+
+func getDurationAndSetLevel(message *fimpgo.Message, outLvlSwitch Service, lvl int64) error {
```

Comment:
This `And` inside the name clearly suggests a function does more than one thing. IMO setting the level can be part of `HandleCmdLvlSet` and this function should be responsible for providing a proper duration.